operations/helm/charts/alloy/values.yaml

```diff
@@ -245,38 +245,83 @@ controller:
   enableStatefulSetAutoDeletePVC: false
 
   autoscaling:
-    # -- Creates a HorizontalPodAutoscaler for controller type deployment.
+    # -- Creates an autoscaler for the controller.
```

Comment:
Should this description say that it uses HPA by default for backwards compatibility?

Reply:
You are definitely right. I also added a comment to explain that this is deprecated and that people should use the other property instead

Reply:
I tried to test this, and it seems the "backwards compatibility" only supports configuring an HPA with default values when using the top level enabled, it doesn't configure it using the config.

Can we do something similar to the `(or .Values.controller.autoscaling.horizontal.enabled .Values.controller.autoscaling.enabled) ` to use the top level config if migrating from a previous helm chart? (And add a test for using the old format)

Reply:
I tried to keep the statefulset autoscaling test with the old values.

I think we could keep the deprecated config support for all the other properties but it gets a bit harder to maintain.

Are we fine that if `.Values.controller.autoscaling.enabled: true` then we use the old configs and if it's `.Values.controller.autoscaling.horizontal.enabled: true` then we use the new config?

The main issue is have is around handling those https://github.com/grafana/alloy/pull/1305/files#diff-e244a7bf61d042b6d261e624ade5f568e6c24851cccc0c2a4666c8bc5f0e3badR28. A helm with is a bit harder to use in that case :)

Reply:
Yeah, if top level enabled we use old support and ignore new, if horizontal.enabled we use horizontal config and ignore old makes sense to me.

No rush on this change, it will most likely wait for @petewall to confirm since he was involved previously and I see he's on PTO this week.